listeners.py

```diff
@@ -178,6 +179,86 @@ def __del__(self):
                 pass
 
 
+class StatisticsMonitor(BaseWatcher):
+    def __init__(self, get_not_done_worker_ids, worker_id_to_pid):
+        self.get_not_done_worker_ids = get_not_done_worker_ids
+        self.worker_id_to_pid = worker_id_to_pid
+        self.worker_current_task = dict()
+        self.failed_tasks = dict()
+        self.rss_results = dict()
+
+    # task name set as test name with running configuration
+    def get_task_name_conf(self, name, conf):
+        if not conf:
+            return name
+        return name + ':' + conf
+
+    def process_result(self, obj):
+        # called only once on task run initiated
+        if isinstance(obj, WorkerCurrentTask):
+            task_name = self.get_task_name_conf(obj.task_name, obj.task_param)
+            self.worker_current_task[obj.worker_id] = obj
+            self.rss_results[task_name] = get_proc_stat_rss(
+                    self.worker_id_to_pid[obj.worker_id])
+
+        # called only once on task run finished
+        if isinstance(obj, WorkerTaskResult):
+            task_name = self.get_task_name_conf(obj.task_id[0], obj.task_id[1])
+            if obj.short_status == 'fail':
+                self.failed_tasks[task_name] = 1
+
+    # called on each 'delta_seconds' delay within task run
+    def process_timeout(self, delta_seconds):
```

Comment:
It is not guaranteed to be called once per 0.1 seconds. We can receive results from the results queue quite often (especially due to #280) and never execute `process_timeout()`. So we should collect metrics in `process_result()` too, but ratelimit it: save when we collect them last time and collect again only after at least 0.1 seconds.